packages/metro/src/DeltaBundler/DeltaTransformer.js

```diff
@@ -108,7 +108,7 @@ class DeltaTransformer extends EventEmitter {
     // different bundles to share the same ids (on HMR, where we need to patch
     // the correct module).
     this._getModuleId = this._bundleOptions.isolateModuleIDs
-      ? createModuleIdFactory()
+      ? (bundleOptions.createModuleIdFactory || createModuleIdFactory)()
```

Comment:
Note that with this, you're going to use the custom `createModuleIdFactory()` only when building prod bundles (for development, `isolateModuleIDs` is usually set to `false`.

That is good, since for development we need to ensure that the same `getModuleId` function instance is used for each `DeltaTransformer` instance (to keep HMR working correctly)